### PR TITLE
Fix product of `[]`

### LINKED ISSUE #978: `Π` returns `0` for an empty list
By mathematical definition, the product of [] = 1.

### PR DESCRIPTION
**Pull request by Steffan153**
Closes #978

### CODE REVIEW

**Review by cgccuser — CHANGES_REQUESTED** [review]
This won't work when the product is *supposed* to be 0

File: vyxal/elements.py [hunk 1]
```diff
@@ -3376,7 +3376,7 @@ def product(lhs, ctx):
     """
 
     if all(vy_type(x) == NUMBER_TYPE for x in lhs):
-        return vy_reduce(multiply, lhs, ctx=ctx)
+        return vy_reduce(multiply, lhs, ctx=ctx) or 1
```
Comment (cgccuser) [inline vyxal/elements.py:3379]:
```suggestion
        return foldl(multiply, lhs, initial=1, ctx=ctx)
```

**Review by cgccuser — CHANGES_REQUESTED** [review]

**Comment by cgccuser** [comment]
Good catch, maybe sum and other elements also need to be fixed?

**Review by cgccuser — APPROVED** [review]

**Review by Lyxal — APPROVED** [review]

File: vyxal/helpers.py [hunk 2]
```diff
@@ -165,9 +165,6 @@ def foldl(
     ctx: Context,
 ) -> Any:
     """Reduce vector by function"""
-    if len(vector) == 0:
-        return 0
```
Comment (chunkybanana) [inline vyxal/helpers.py:169]: This breaks other elements that use foldl.

**Review by chunkybanana — CHANGES_REQUESTED** [review]

File: vyxal/helpers.py (hunk 2 above)
Comment (Steffan153) [reply vyxal/helpers.py:169]: It doesn't, notice the `initial=None`, and if `initial` is `None`, it defaults to `0`. (I don't know why user didn't just do `initial=0`).

Comment (cgccuser) [reply vyxal/helpers.py:169]: The fault is in those other elements, not `foldl`. They should give an initial value or check for empty lists themselves.

**Pull request merged**